src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -3109,19 +3109,30 @@ else if (type == RStudioApiRequestEvent.TYPE_DOCUMENT_OPEN)
             @Override
             public void onSuccess(final EditingTarget result)
             {
-               if (data.getRow() != -1) 
+               Command finish = () -> {
+                  // suppress attempts by the shell widget to steal focus here
+                  events_.fireEvent(new SuppressNextShellFocusEvent());
+               
+                  result.focus();
+                  result.ensureCursorVisible();
+                  
+                  JsObject response = JsObject.createJsObject();
+                  response.setString("id", result.getId());
+                  server_.rstudioApiResponse(response, new VoidServerRequestCallback());
+               };
+
+               int row = data.getRow();
+               if (row != -1) 
                {
-                  columnManager_.scrollToPosition(FilePosition.create(data.getRow(), data.getColumn()), data.getMoveCursor());
+                  // give ace time to render before scrolling to position
+                  Scheduler.get().scheduleDeferred(() ->
```

Comment:
I'm not sure about this, but it seems to do the trick. Without this, the ace renderer is not ready, and e.g. has a `lineheight` of `0` which does not play well with this: 

```js
    this.scrollToRow = function(row) {
        this.session.setScrollTop(row * this.lineHeight);
    };
```

Reply:
This looks fine to me.